_includes/app/txn-sample.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/lib/pq"
+)
+
+func transferFunds(tx *sql.Tx, from int, to int, amount float64) error {
+	// Read the balance.
+	var fromBalance float64
+	if err := tx.QueryRow(
+		"SELECT balance FROM accounts WHERE id = $1", from).Scan(&fromBalance); err != nil {
+		return err
+	}
+
+	if fromBalance < amount {
+		return fmt.Errorf("insufficient funds")
+	}
+
+	// Transfer $100 to the other account.
+	if _, err := tx.Exec(
+		"UPDATE accounts SET balance = balance - $1 WHERE id = $2", amount, from); err != nil {
+		return err
+	}
+	if _, err := tx.Exec(
+		"UPDATE accounts SET balance = balance + $1 WHERE id = $2", amount, to); err != nil {
+		return err
+	}
+	return nil
+}
+
+// txnWrapper runs fn inside a transaction and retries it as needed.
+func txnWrapper(db *sql.DB, database string, fn func(*sql.Tx) error) error {
+	// Start a transaction.
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	// Specify that we intend to retry this txn in case of CockroachDB retriable
+	// errors.
+	if _, err := tx.Exec("SAVEPOINT cockroach_restart"); err != nil {
+		return err
+	}
+	if _, err := tx.Exec("SET DATABASE = " + database); err != nil {
```

Comment:
Do we need this here? can we just connect to the right DB in the conn string to?